src/main/scala/tech/sourced/gemini/ReportApp.scala

```diff
@@ -166,4 +200,25 @@ object ReportApp extends App {
     writerBuckets.close()
   }
 
+  def readCommunities(dirPath: String): List[(Int, List[Int])] = {
+    val parquetFilePath = new Path(s"$dirPath/communities.parquet")
+    val reader = AvroParquetReader.builder[GenericRecord](parquetFilePath).build()
+
+    Iterator
+      .continually(reader.read)
+      .takeWhile(_ != null)
+      .map(record => {
```

Comment:
Very minor, but usually in such cases just

```scala
.map { record => 
  // do stuff
}
.toList
```

is enough.

Reply:
reader doesn't have `map`. I saw this pattern with `Iterator.continually.takeWhile` in engine and jgit a lot. So I think we can keep it as is.

Reply:
:+1: there is just extra `()` that could be avoided here